libjolie/src/jolie/lang/nativeTypes/NativeType.java

```diff
@@ -28,45 +28,80 @@
  *
  * @author Fabrizio Montesi
  */
-public enum NativeType
-{
-	//UNDEFINED( "undefined" ),
-	STRING( "string" ),
-	INT( "int" ),
-	LONG( "long" ),
-	BOOL( "bool" ),
-	DOUBLE( "double" ),
-	VOID( "void" ),
-	RAW( "raw" ),
-	ANY( "any" );
-
-	private final static Map< String, NativeType > idMap = new HashMap<>();
-
-	static {
-		for( NativeType type : NativeType.values() ) {
-			idMap.put( type.id(), type );
-		}
+public class NativeType {
+
+	String refinement = "";
+	NativeTypeEnum type;
+
+	public NativeType(NativeTypeEnum type) {
+		this.type = type;
+	}
+
+	public NativeType(String refinement, NativeTypeEnum type) {
+		this.refinement = refinement;
+		this.type = type;
+	}
+
+	public NativeType() {
 	}
 
-	private final String id;
-	
-	private NativeType( String id )
-	{
-		this.id = id;
+	public NativeTypeEnum getType() {
+		return type;
 	}
 
-	public String id()
-	{
-		return id;
+	public void setType(NativeTypeEnum type) {
```

Comment:
What's the usage for this? Can't we remove it? The type should be set in the constructor.